### ISSUE TITLE
Custom release config file for custom packages

### ISSUE THREAD

**Issue by cyberblast** [microsoft/appcenter#1259]
**Describe the solution you'd like**
Custom release config file within custom zip archive, like 'appcenter.json' containing
* version number
* release notes
* path to app icon within zip file or embedded base64 icon

**Describe alternatives you've considered**
Upload of custom zip archive release using api.

**Additional context**
Hi, i already managed to utilize rest api methods with MS Flow to publish completely custom packages, zipped with Azure DevOps pipes (the DevOps "AppCenter distribute" action is not working for our custom packages - no apk/aab/ipa).
However, there are several things I'm missing and it would be much easier to generate all those relase information during the build and to write it to a file and include it with the build/package. 

1. Actually I have no access to our custom build number within flow (from package.json). I'd need to explicitly access the repo and read it. Thus we are actually publishing packages in AppCenter using the instance id of the release pipeline as version number. It would be much easier to ship that information within the package.
2. Same with release notes - currently I'm updating them manually to the app release after mails have been triggered by the publishing flow.
3. I manually changed the app icon but it is not showing on the install portal.

**Comment by Oddj0b** [comment]
Thank you for your great description. Which platform are you distributing and building?

**Comment by cyberblast** [comment]
Well, one of the packages includes different installers for a distributed system containing an "xcopy" package for the frontend (angular), a web deploy package for the middleware (.Net WebApi), a sharepoint solution, WebCon Workflow artifacts, SQL DB scripts and a lof of powershell etc. 
It's not an "app" in the strict sense. 
But these artifacts always need to go together and are all belonging to a joint single distribution version/package each as there are a lot of dependencies. 

An other package is just a bunch of scripts, presenting something like a dev or operations tool. But we are packaging and versioning that as well. 

That's why I'm talking about "custom zip archive" :) 

However, in the meanwhile I developed an azure function to strip a single file out of a zip package to get a custom manifest.json containing more required metadata for the distribution process to appcenter (flow/appcenter api). So I solved most of the issues myself already. The icon topic is not covered of course. 
Nevertheless I still believe the initial idea would be a nice feature addition for app center's support of custom packages.
